### PR TITLE
Add initial mac testing on CircleCI

### PR DESCRIPTION
**Pull request by sbc100**

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -187,12 +265,12 @@ jobs:
       - run:
           name: install emsdk
           command: |
-            wget -O ~/emsdk-master.tar.gz https://github.com/juj/emsdk/archive/master.tar.gz
+            curl -# -L -o ~/emsdk-master.tar.gz https://github.com/emscripten-core/emsdk/archive/master.tar.gz
```
Comment (kripken) [inline .circleci/config.yml]: why curl?

Comment (sbc100) [reply .circleci/config.yml]: Because the macOS didn't come with wget preinstalled.  I guess I could have installed it with brew but this seemed easier.

**Comment by sbc100** [comment]
`other` is just the starting point.  We can ramp it up over time.

**Comment by sbc100** [comment]
I will add wasm2 as well actually

**Comment by sbc100** [comment]
Do you know how I configure which circleci steps are "Required"?

**Comment by kripken** [comment]
The required steps are in the admin section of the github UI for the repo, `Settings => Branches => Branch Protection Rules`.

**Comment by sbc100** [comment]
> The required steps are in the admin section of the github UI for the repo, `Settings => Branches => Branch Protection Rules`.

I share that concern.  Thats why I'm limiting the number of tests we run on mac to a small subset.

**Pull request merged**